sys/core/src/TxsCore.hs

```diff
@@ -172,16 +173,25 @@ import VarId
 -- | TorXakis core main api -- start
 runTxsCore :: Config -> StateT s IOC.IOC a -> s -> IO ()
 runTxsCore initConfig ctrl s0  =  do
-     _ <- runStateT (runTxsCtrl ctrl s0)
-               IOC.EnvC { IOC.config = initConfig
-                         , IOC.unid   = 0
-                         , IOC.params = initParams
-                         , IOC.state  = initState
-                         }
-     return ()
-       where initState = IOC.Noning
-             initParams =
-               Map.union ParamCore.initParams SolveDefs.Params.initParams
+      _ <- runStateT (runTxsCtrl ctrl s0)
+              IOC.EnvC { IOC.config = initConfig
+                        , IOC.unid   = 0
+                        , IOC.params = updateParams initParams
+                                        $ Config.configuredParameters initConfig
+                        , IOC.state  = initState
+                        }
+      return ()
+      where initState = IOC.Noning
+            initParams =
+              Map.union ParamCore.initParams SolveDefs.Params.initParams
+
+updateParams :: ParamCore.Params -> [(Config.ParameterName,Config.ParameterValue)] -> ParamCore.Params
+updateParams = foldl applyConfParam
+  where applyConfParam allParams (Config.ParamName pnStr, Config.ParamValue pvStr) =
+          Map.adjust (updateOldParamWith pvStr) ("param_" ++ pnStr) allParams
```

Comment:
Something that might be a source of headaches in the future is that the knowledge that we have to prepend `"param_" to the parameter name is hidden here. Isn't the `Config` module a better place for it? You could make the accessor function private, and export a `getTxsParamName` (or something like that) which will perform this transformation. In that way we might prevent errors in the future.

Reply:
Moved to Config, looks like it's not needed anymore since this is now internal in Config